nucypher/blockchain/eth/utils.py

```diff
@@ -159,23 +161,22 @@ def get_default_rpc_endpoints() -> Dict[int, List[str]]:
         return {}
 
 
-def get_healthy_default_rpc_endpoints(chain_id: int) -> List[str]:
-    """Returns a list of healthy RPC endpoints for a given chain ID."""
+def get_healthy_default_rpc_endpoints(domain: TACoDomain) -> Dict[int, List[str]]:
+    """Returns a mapping of chain id to healthy RPC endpoints for a given domain."""
 
-    endpoints = get_default_rpc_endpoints()
-    chain_endpoints = endpoints.get(chain_id)
+    endpoints = get_default_rpc_endpoints(domain)
 
-    if not chain_endpoints:
-        LOGGER.error(f"No default RPC endpoints found for chain ID {chain_id}")
-        return list()
-
-    healthy = [
-        endpoint for endpoint in chain_endpoints if rpc_endpoint_health_check(endpoint)
-    ]
-    LOGGER.info(f"Healthy default RPC endpoints for chain ID {chain_id}: {healthy}")
-    if not healthy:
-        LOGGER.warn(
-            f"No healthy default RPC endpoints available for chain ID {chain_id}"
-        )
+    if not domain.is_testnet:
```

Comment:
because there are soooo many endpoints that it will be checking. This will take a very long time on testnet

Reply:
^ Just to add to what @theref said, it would involve doing health checks on all of these URLs - https://github.com/nucypher/chainlist/blob/main/lynx.json, which seems tedious and excessive for `lynx` which seems excessive.

If we expand `mainnet` to support "any" evm chain, it's something we have to consider possibly not doing. Given that we fallback to the next RPC URL in the list of URLs of a previous endpoint fails, the health check may be excessive at that point. We can adjust the list in realtime as certain URLs work so that the list is biased over time...?